ctx.go

```diff
@@ -220,6 +223,67 @@ func (c netHttpContext[B]) PathParam(name string) string {
 	return c.Req.PathValue(name)
 }
 
+type PathParamNotFoundError struct {
+	ParamName string
+}
+
+func (e PathParamNotFoundError) Error() string {
+	return fmt.Errorf("param %s not found", e.ParamName).Error()
+}
+
+type PathParamInvalidTypeError struct {
+	Err          error
+	ParamName    string
+	ParamValue   string
+	ExpectedType string
+}
+
+func (e PathParamInvalidTypeError) Error() string {
+	return fmt.Errorf("param %s=%s is not of type %s: %w", e.ParamName, e.ParamValue, e.ExpectedType, e.Err).Error()
+}
+
+type ContextWithPathParam interface {
+	PathParam(name string) string
+}
+
+func PathParamIntErr(c ContextWithPathParam, name string) (int, error) {
+	param := c.PathParam(name)
+	if param == "" {
+		return 0, PathParamNotFoundError{ParamName: name}
+	}
+
+	i, err := strconv.Atoi(param)
+	if err != nil {
+		return 0, PathParamInvalidTypeError{
```

Comment:
Discussion. Maybe we should return a fuego.BadRequest error here?

Reply:
Yeah directly, could be nice. I think he tried to mimic what's in `common_context.go`. I propose to merge this MR and refactor in the 2 places after, and keep compatibility by relying on tests.

Reply:
I agree. Resolving